#! /bin/sh /usr/share/dpatch/dpatch-run
## fix_set_encoding.dpatch by Miguel Landaeta <miguel@miguel.cc>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: method setCharEncoding from Tidy class was removed from API.
## DP: now the methods setInputEncoding and setOutputEncoding must be used.

@DPATCH@
diff -urNad httpunit-1.7+dfsg~/src/com/meterware/httpunit/parsing/JTidyHTMLParser.java httpunit-1.7+dfsg/src/com/meterware/httpunit/parsing/JTidyHTMLParser.java
--- httpunit-1.7+dfsg~/src/com/meterware/httpunit/parsing/JTidyHTMLParser.java	2008-05-20 16:25:36.000000000 +0000
+++ httpunit-1.7+dfsg/src/com/meterware/httpunit/parsing/JTidyHTMLParser.java	2009-03-04 21:25:54.000000000 +0000
@@ -87,7 +87,8 @@
 
     private static Tidy getParser( URL url ) {
         Tidy tidy = new Tidy();
-        tidy.setCharEncoding( org.w3c.tidy.Configuration.UTF8 );
+        tidy.setInputEncoding( "UTF8" );
+        tidy.setOutputEncoding( "UTF8" );
         tidy.setQuiet( true );
         tidy.setShowWarnings( HTMLParserFactory.isParserWarningsEnabled() );
         if (!HTMLParserFactory.getHTMLParserListeners().isEmpty()) {
